cli/cmd/root.go

```diff
@@ -102,18 +106,38 @@ func Execute(stdin io.Reader, stdout io.Writer, stderr io.Writer) error {
 				return errors.New("Please provide your API token")
 			}
 		}
-		api := client.NewHTTPClient(apiOrigin, apiToken)
-		runCmds.api = api
+		platformAPI := platform_client.NewHTTPClient(platformOrigin, apiToken)
+		runCmds.platformAPI = platformAPI
+
+		shipAPI := ship_client.NewGraphQLClient(shipOrigin, apiToken)
+		runCmds.shipAPI = shipAPI
+
+		commonAPI := client.NewClient(platformOrigin, shipOrigin, apiToken)
+		runCmds.api = commonAPI
 
 		if appSlugOrID == "" {
 			appSlugOrID = os.Getenv("REPLICATED_APP")
 		}
 
-		app, err := api.GetApp(appSlugOrID)
+		appType, err := commonAPI.GetAppType(appSlugOrID)
 		if err != nil {
 			return err
 		}
-		runCmds.appID = app.Id
+		runCmds.appType = appType
+
+		if appType == "platform" {
```

Comment:
yeah this feels like the kind of thing that would benefit from an interface but I don't really see a clear path to getting there yet

Reply:
agreed.. i'm going to add some more implementation to ship methods, and look for opportunity for a pattern to emerge